### PR TITLE
Use `"format": "uuid"` instead of a regex

### PR DESCRIPTION
**Pull request by thejcannon**
From https://github.com/buildkite/pipeline-schema/pull/102#discussion_r1864988907

### CODE REVIEW

**Review by DrJosh9000 — APPROVED** [review]
LGTM thank you

**Comment by akshayhacholli** [comment]
Can we revert this change? Our schema validations are failing and the step key parameter has a non-uuid value.

**Comment by akshayhacholli** [comment]
Thanks Josh for reverting. I think the problem is while we are using python3-jsonschema, I am unable to get a working failure to share here, but give me some time. If I can't produce a repro failure, you can merge these changes later.

**Comment by akshayhacholli** [comment]
Was able to repro the failure with this python script - 
* using python3
* installed pip3 libraries `pyyaml` and `jsonschema`
* with uuid change the script fails and succeeds with the revert

```
import os
import json
import jsonschema
from yaml import Loader, load


def main():
    yaml_extensions = [".yml", ".yaml"]
    with open('schema.json', 'r') as file:
        schema = json.load(file)
    for path, currentDirectory, files in os.walk("test/valid-pipelines"):
        for f in files:
            if os.path.splitext(f)[1] in yaml_extensions:
                file = os.path.join(path, f)
                with open(file, "r") as pipeline_file:
                    yaml = load(pipeline_file.read(), Loader)
                    jsonschema.validate(instance=yaml, schema=schema)


main()

```

**Comment by rogpeppe** [comment]
FWIW the failure might be to do with the fact that implementations [are not required to implement the `format` keyword](https://json-schema.org/draft-07/draft-handrews-json-schema-validation-01#rfc.section.7.2). In later JSON Schema versions it's not even a validation keyword, but [just an annotation](https://json-schema.org/draft/2019-09/release-notes).

For an implementation that does not implement the `format` keyword (or does not implement `uuid` specifically) then `"not": {"format": "uuid"}` will never pass validation, so it's probably not what you want.

In other words, I think the regexp validation is the better approach.

I came to the previous issue to say this but was too late, sorry! :)
